packages/core/core/src/services/document-service/transform/fields.ts

```diff
@@ -1,13 +1,33 @@
-export const transformFields = (fields: string[]): string[] => {
-  if (!fields || fields.length === 0 || !Array.isArray(fields)) {
+import { Documents, Common } from '@strapi/types';
+
+type Fields = Documents.Params.Pick<Common.UID.Schema, 'fields'>['fields'];
+
+export const transformFields = (fields: Fields): Fields => {
+  // If  it's a string, and it doesn't contain documentId, should be an array
+  // '*' => '*'
+  // 'documentId' => 'documentId' , 'documentId,name' => 'documentId,name'
+  // 'name' => ['name', 'documentId']
+  // 'id' => 'documentId'
```

Comment:
@jhoward1994 unrelated to this pr but realised there were a couple of cases we were not covering here!

Reply:
Thank you!